xcdat/averager.py

```diff
@@ -0,0 +1,501 @@
+        Axis orientation conversion type.
+
+    Returns
+    -------
+    RegionAxisBounds
+        List of converted longitude values.
+
+    Notes
+    -----
+    This does not re-order the values in any way; it only converts the values
+    in-place between longitude conventions (-180 to 180) or (0 to 360).
+    """
+    rl_nparray = np.array(region_lon)
+
+    if to == "360":
+        # TODO: Add handle if lon isn't 180 to begin with
+        inds = np.where(rl_nparray < 0)[0]
+        rl_nparray[inds] = rl_nparray[inds] + 360
+    elif to == "180":
+        # TODO: Add handle if lon isn't 360 to begin with
+        inds = np.where(rl_nparray > 180)[0]
+        rl_nparray[inds] = rl_nparray[inds] - 360
+
+    return rl_nparray.tolist()
+
+
+def _is_region_lon_in_domain(
+    region_bounds: RegionAxisBounds, domain_bounds: xr.DataArray
+) -> bool:
+    """Check if a region longitude bounds fall within domain longitude bounds.
+
+    Parameters
+    ----------
+    region_bounds : RegionAxisBounds
+        Region longitude bounds.
+    domain_bounds : xr.DataArray
+        Domain longitude bounds.
+
+    Returns
+    -------
+    bool
+        True if all region longitude bounds are within grid longitude bounds.
+    """
+    min_domain_bounds = domain_bounds.min(axis=1)
+    max_domain_bounds = domain_bounds.max(axis=1)
+
+    in_domain = True
+    for bound in region_bounds:
+        count_in_domain = np.where(
+            (min_domain_bounds <= bound) & (max_domain_bounds >= bound)
+        )[0]
+        if count_in_domain.size == 0:
+            in_domain = False
+
+    return in_domain
+
+
+def _get_axis_weights(
+    xvariable: xr.DataArray,
+    lat_bounds: Optional[RegionAxisBounds] = None,
+    lon_bounds: Optional[RegionAxisBounds] = None,
+    is_weighted: bool = True,
+) -> AxisWeights:
+    """
+    Function to generate axis weights for subsequent operations (e.g.,
+    area averaging) with the option to generate weights based on a rectilinear
+    regional selection.
+
+    Operations include:
+
+    - Generate weights for axes using dimension bounds.
+       - If dimension doesn't have bounds, weights aren't calculated for it.
+    - If a region is specified, calculate the weights corresponding to the
+       regional bounds.
+    - Latitude weights are scaled with sine(latitude).
+
+    Parameters
+    ----------
+    xvariable : xr.DataArray
+        xr.DataArray produced using ``open_variable``
+    lat_bounds : Optional[RegionAxisBounds], optional
+        List of latitude boundaries for regional selection (e.g.,
+        lat_bounds=[-5, 5]), by default None
+    lon_bounds : Optional[RegionAxisBounds], optional
+        List of longitude boundaries for regional selection (e.g.,
+        lon_bounds=[-170, -120]), by default None
+    is_weighted : bool, True
+        Weighting option to calculate weights proportional to area
+        (True), uniform weights (False).
+
+    Returns
+    -------
+    AxisWeights
```

Comment:
These are Python type aliases, which are defined at the top of the module. `AxisWeights = Dict[Hashable, xr.DataArray]`

The reason I prefer type aliases is because it is a reusable. Type aliases are documented in the xCDAT docs.
We can also generate a tuple of options using the type alias if it is a `Literal`, which is great for input validation.

Example:
```python
#: List of supported axis for spatial averaging
SupportedAxes = Literal["lat", "lon"]
SUPPORTED_AXES: Tuple[SupportedAxes, ...] = get_args(SupportedAxes)

if axis not in SUPPORTED_AXIS:
   ....
```